app/models/service_order.rb

```diff
@@ -17,6 +17,12 @@ class ServiceOrder < ApplicationRecord
   before_create :assign_user
   after_create  :create_order_name
 
+  def initialize(*args)
+    raise NotImplementedError, _("must be implemented in a subclass") if self.class == ServiceOrder
```

Comment:
@bdunne I added this to protect the base class.  Do you know if there is a better solution?

I tried using `private_class_method :new` but that caused lots of test failures.